slasher/detection/detect.go

```diff
@@ -170,6 +170,30 @@ func (ds *Service) DetectDoubleProposals(ctx context.Context, incomingBlock *eth
 	return ds.proposalsDetector.DetectDoublePropose(ctx, incomingBlock)
 }
 
+// mapResultsToAtts handles any duplicate detections by ensuring they reuse the same pool of attestations, instead of re-checking the DB for the same data.
+func (ds *Service) mapResultsToAtts(ctx context.Context, results []*types.DetectionResult) (map[[32]byte][]*ethpb.IndexedAttestation, error) {
+	ctx, span := trace.StartSpan(ctx, "detection.mapResultsToAtts")
+	defer span.End()
+	resultsToAtts := make(map[[32]byte][]*ethpb.IndexedAttestation)
+	for _, result := range results {
+		resultKey := resultHash(result)
+		if _, ok := resultsToAtts[resultKey]; ok {
+			continue
+		}
+		matchingAtts, err := ds.slasherDB.IndexedAttestationsWithPrefix(ctx, result.SlashableEpoch, result.SigBytes[:])
+		if err != nil {
+			return nil, err
+		}
+		resultsToAtts[resultKey] = matchingAtts
+	}
+	return resultsToAtts, nil
+}
+
+func resultHash(result *types.DetectionResult) [32]byte {
+	resultBytes := append(bytesutil.Bytes8(result.SlashableEpoch), result.SigBytes[:]...)
+	return hashutil.FastSum256(resultBytes)
```

Comment:
Are collisions OK? This is known unsafe

Reply:
Collisions are fine but maybe I should change it to sha256, not very expensive anyways.